packages/web-ui/src/utils/with-breakpoints.ts

```diff
@@ -0,0 +1,20 @@
+import { Breakpoints, Responsive } from '../types';
+import { getPrefixedName } from './utils';
+
+export const withBreakpoints = (value: Responsive<string> | undefined, prefix = '') => {
+  if (typeof value === 'string') {
+    return getPrefixedName(`${prefix}-${value}`);
+  }
+
+  if (typeof value === 'object') {
+    const classes: string[] = [];
+    for (const bp of Object.keys(value) as Breakpoints[]) {
+      if (bp in value) {
+        const baseClassName = getPrefixedName(`${prefix}-${value[bp]}`);
+        const className = bp === 'mobile' ? baseClassName : `${bp}:${baseClassName}`;
+        classes.push(className);
+      }
+    }
```

Comment:
any reason to not use `Array.map` here? I think it would be cleaner

Reply:
Nope, updated 👍🏻